### PR TITLE
SERVICES-1379: compute undistributedBoostedRewards

### PR DESCRIPTION
**Pull request by dragos-rebegea**
compute undistributedBoostedRewards including unclaimed rewards

## Reasoning
- Currently undistributedBoostedRewards only returns the amounts that has been collected using `collectUndistributedBoostedRewards`. Since undistributedBoostedRewards should contains also the uncollected, we should compute the value.
  
## Proposed Changes
- Calculates the total undistributed boosted rewards for a given smart contract address (scAddress) up to the current week (currentWeek) by first fetching the undistributed boosted rewards and the last week when these rewards were collected. If the period of unclaimed weeks is more than the maximum number of weeks a user can claim, it only considers the rewards from the last USER_MAX_CLAIM_WEEKS. It then calculates the rewards remaining to be distributed for each week starting from the week after the last collection up to the week derived from subtracting the maximum claim weeks from the current week. The total of these remaining rewards is then added to the previously fetched undistributed boosted rewards to get the total amount.

## How to test
- Test by providing a smart contract address (scAddress) and the current week (currentWeek). Check that the function returns the correct total undistributed boosted rewards.
- Test the edge case where the starting week is later than the ending week, the function should return the undistributed boosted rewards without calculating the remaining rewards.

### CODE REVIEW

File: src/modules/farm/v2/services/farm.v2.compute.service.ts [hunk 1]
```diff
@@ -457,6 +463,54 @@ export class FarmComputeServiceV2
...
+            promises.push(
+                this.farmAbi.remainingBoostedRewardsToDistribute(
+                    scAddress,
+                    week,
+                )
+            )
+        }
+        const remainingRewards = await Promise.all(promises);
+        const totalRemainingRewards = remainingRewards.reduce((acc, curr) => {
+            return new BigNumber(acc).plus(curr).integerValue().toFixed();
```
Comment (claudiulataretu) [inline src/modules/farm/v2/services/farm.v2.compute.service.ts:505]: you can return BigNumber here

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/farm/specs/farm.v2.compute.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }).compile();
+    });
+    it("should correctly calculate total rewards", async () => {
+        const service = module.get<FarmComputeServiceV2>(
+            FarmComputeServiceV2,
+        );
+        const result = await service.computeUndistributedBoostedRewards('erd18h5dulxp5zdp80qjndd2w25kufx0rm5yqd2h7ajrfucjhr82y8vqyq0hye', 10);
+        const expectedTotal = new BigNumber('5000').plus('4000').integerValue().toFixed(); // 4 weeks * 1000
+        expect(result).toEqual(expectedTotal);
+        // expect(mockFarmAbi.undistributedBoostedRewards).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply src/modules/farm/specs/farm.v2.compute.service.spec.ts:67]: I let them for me for future reference to be implemented when it will be possible.

**Review by claudiulataretu — DISMISSED** [review]

File: src/modules/farm/v2/services/farm.v2.getter.service.ts
Comment (claudiulataretu) [inline src/modules/farm/v2/services/farm.v2.getter.service.ts:1]: remove empty file

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

**Review by claudiulataretu — APPROVED** [review]
